Competition/src/main/java/competition/subsystems/elevator/ElevatorSubsystem.java

```diff
@@ -0,0 +1,133 @@
+import xbot.common.controls.sensors.XDigitalInput;
+import xbot.common.injection.wpi_factories.CommonLibFactory;
+import xbot.common.math.MathUtils;
+import xbot.common.math.PID;
+import xbot.common.math.PIDFactory;
+import xbot.common.properties.XPropertyManager;
+
+@Singleton
+public class ElevatorSubsystem extends BaseSubsystem {
+    private static Logger log = Logger.getLogger(ElevatorSubsystem.class);
+
+    public XCANTalon master;
+    public XDigitalInput calibrationSensor;
+    private double calibrationSensorPosition = -1;
+    private boolean isCalibrated;
+    private double power;
+
+    public enum HatchLevel {
+        Low, Medium, High,
+    }
+
+    @Inject
+    public ElevatorSubsystem(CommonLibFactory factory, XPropertyManager propManager, ElectricalContract2019 contract,
+            PIDFactory pd) {
+        log.info("Creating ElevatorSubsystem");
+        this.master = factory.createCANTalon(contract.getElevatorMasterMotor().channel);
+        master.configSelectedFeedbackSensor(FeedbackDevice.QuadEncoder, 0, 0);
+        this.calibrationSensor = factory.createDigitalInput(contract.getElevatorCalibrationSensor().channel);
+    }
+
+    public void stop() {
+        setPower(0);
+    }
+
+    public void raiseElevator() {
+        setPower(1);
+    }
+
+    public void lowerElevator() {
+        setPower(-1);
+    }
+
+    // Set's isCalibrated to true is sensor button is pressed
+    public boolean isCalibrationSensorPressed() {
+        return calibrationSensor.get();
+    }
+
+    public boolean calibrationSensorState() {
+        if (isCalibrationSensorPressed()) {
+            isCalibrated = true;
+        }
+        return isCalibrated;
+    }
+
+    public void lowerElevatorToCalibrationPoint() {
+        if (isCalibrated == false) {
+            lowerElevator();
+        }
+    }
+
+    // Returns the tick that the elevator has
+    public double getCurrentTickHeight() {
+        return master.getSelectedSensorPosition(0);
+    }
+
+    // Sets the Calibration Sensor's position at where it is.
+    public void calibrate() {
+        calibrationSensorPosition = getCurrentTickHeight();
+    }
+
+    // Returns the Calibration Sensor's height.
+    public double getCalibrationHeight() {
+        return calibrationSensorPosition;
+    }
+
+    private void setPower(double power) {
+        if (isCalibrationSensorPressed()) {
+            power = MathUtils.constrainDouble(power, 0, 1);
+            calibrate();
+        }
+        master.simpleSet(power);
+    }
+
+    public double getPower() {
+        return power;
+    }
+
+    private double getTickHeightForLevel(HatchLevel level) {
+        getCalibrationHeight();
+        double distanceBetweenLevels = -1;
+        if (level == HatchLevel.Low) {
+            return calibrationSensorPosition + distanceBetweenLevels;
+        } else if (level == HatchLevel.Medium) {
+            return calibrationSensorPosition + distanceBetweenLevels * 2;
```

Comment:
Same here - level 2 should be distanceBetweenLevels, level 3 should be distancebetweenLevels*2